### ISSUE TITLE
Graphite: Fetching metrics failed: undefined

### ISSUE THREAD

**Issue by nikobearrr** [grafana/grafana#45947]
**What happened**:

When trying to browse through graphite metrics a red popup shows saying `Fetching metrics failed: undefined`

**What you expected to happen**:

I expect not to see any error

**How to reproduce it (as minimally and precisely as possible)**:

1. Go to play.grafana.org
2. Add some graphite panel/series
3. type #A 

(see attached screenshot)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.1
- Data source type & version: graphite which is running on play.grafana.org
- User OS & Browser: macOS / chrome

![image](https://user-images.githubusercontent.com/8693064/155884769-76d75f8a-d5b6-415b-8b4c-47c8717d37c5.png)

console log:

```
{type: 'cancelled', cancelled: true, data: null, status: -1, statusText: 'Request was aborted', …}cancelled: trueconfig: {method: 'POST', url: 'api/datasources/proxy/1/metrics/find', params: {…}, data: 'query=*#*', headers: {…}, …}data: nullstatus: -1statusText: "Request was aborted"type: "cancelled"[[Prototype]]: Object
(anonymous) @ instrument.js:109
overrideMethod @ react_devtools_backend.js:4061
Qe @ helpers.ts:175
ot @ providers.ts:96
await in ot (async)
(anonymous) @ providers.ts:111
(anonymous) @ MetricSegment.tsx:31
h @ lodash.js:10401
g @ lodash.js:10411
j @ lodash.js:10478
(anonymous) @ useAsyncFn.js:19
(anonymous) @ useAsync-6eaa3052.esm.js:89
(anonymous) @ useAsync-6eaa3052.esm.js:133
(anonymous) @ useStateManager-783b07d5.esm.js:49
value @ Select-09b3775e.esm.js:1891
r.handleInputChange @ Select-09b3775e.esm.js:1567
Ue @ react-dom.production.min.js:52
Ke @ react-dom.production.min.js:52
(anonymous) @ react-dom.production.min.js:53
kr @ react-dom.production.min.js:100
Er @ react-dom.production.min.js:101
(anonymous) @ react-dom.production.min.js:113
Ne @ react-dom.production.min.js:292
(anonymous) @ react-dom.production.min.js:50
Ar @ react-dom.production.min.js:105
Jt @ react-dom.production.min.js:75
Xt @ react-dom.production.min.js:74
t.unstable_runWithPriority @ scheduler.production.min.js:18
Ho @ react-dom.production.min.js:122
ze @ react-dom.production.min.js:292
$t @ react-dom.production.min.js:73
o @ helpers.js:73
```

![image](https://user-images.githubusercontent.com/8693064/155884882-6056c957-fba2-40ea-ad97-6114305f0ee2.png)

**Comment by ashharrison90** [comment]
hi @nikobearrr 👋 

i can't reproduce this on play. are you still seeing the error and are there any more specific reproduction steps you can provide?

thanks! 😄

**Comment by ashharrison90** [comment]
thanks @nikobearrr 🙌  i was able to reproduce on https://play.grafana.org/d/cvDFGseGz/graphite-templated-nested?orgId=1&editPanel=1, although it's really intermittent for me. seems to be some sort of timing issue. 🤔 

i'll forward on to the relevant squad and we'll investigate.

**Comment by gtk-grafana** [comment]
I was unable to reproduce this on main (local and cloud), or on 8.4.1 locally.

**Comment by nikobearrr** [comment]
I just reproduced it in the same way on play.grafana.org which is running v9.0.5
Let me know if you want/need a video

**Comment by bohandley** [comment]
Hi all, @nikobearrr I was able to reproduce it with the link you sent, but it required me to type a lot of characters into the input that did not align with any normal graphite queries. You were able to reproduce it with something more simple, the ref-id query in your video, #A, and the characters 'asd'. This leads me to think there may  be a sequence of key strokes, shift, paste, delete or something that may affect something. @gtk-grafana, any thoughts on this?

<img width="1159" alt="Screen Shot 2022-08-08 at 11 52 18 AM" src="https://user-images.githubusercontent.com/25674746/183461890-f7920b30-1b11-4d49-b0f6-7abc34043882.png">

**Comment by gtk-grafana** [comment]
Yeah if I add a spam a bunch of chars I can trigger a similar error, but it's still very intermittent, and it usually appears to happen after the initial page-load, so I'd bet on a race condition. I'll see if I can dig up some more information and debug this locally.

**Comment by gtk-grafana** [comment]
So this is made much more easily reproducible when throttling network speed.
The error appears to be thrown when we're cancelling an ongoing query after the user enters additional input.

So as I enter in `#C` into the query box, we fire off two queries:
first query:
POST: `https://play.grafana.org/api/datasources/proxy/1/metrics/find?from=1659976837&until=1660063245`
Payload: `query: *#*`

Second query:
POST: `https://play.grafana.org/api/datasources/proxy/1/metrics/find?from=1659976836&until=1660063246`
Payload: `query: *#C*`

The error that we're seeing being logged here appears to be thrown from the cancelled API request (`query: *#*`), so maybe the solution for this issue is to make sure that we don't throw errors in the UI when the client application cancels a request to the backend.

**Comment by gtk-grafana** [comment]
Ok so after discovering I could reliably reproduce this issue when throttling network request speed, I found that I can reliably reproduce this on [9.0.6](https://github.com/grafana/grafana/releases/tag/v9.0.6), but not main or [9.1.0-beta1](https://github.com/grafana/grafana/releases/tag/v9.1.0-beta1), and I think @bohandley just released a bunch of changes for graphite in v9.1.0, so it's looking like this issue was resolved already with those changes!

**Comment by gtk-grafana** [comment]
@nikobearrr Are you able to test this locally against the 9.1.0-beta1 branch and verify my result? If not we can wait until 9.1.0 is released (tentatively next week, but it'll take a bit longer to release to the play.grafana.com), and then close this once you can verify that this is no longer an issue.

**Comment by gtk-grafana** [comment]
The fix for this was released in [v9.1.0](https://github.com/grafana/grafana/releases/tag/v9.1.0). Closing this ticket, but let us know if you're still seeing issues.
